packages/web/src/components/ui/heading.tsx

```diff
@@ -10,27 +11,51 @@ export function EditableH1(props: {
   className?: string;
   onUpdated: (text: string) => void;
 }) {
-  const ref = React.useRef<HTMLHeadingElement | null>(null);
+  const ref = React.useRef<HTMLTextAreaElement | null>(null);
+  const [heading, setHeading] = React.useState<string>(props.text);
+  const [isMaxHeadingLengthExceeded, setIsMaxHeadingLengthExceeded] =
+    React.useState<boolean>(false);
+  const maxHeadingLength = 50;
+
+  const handleChange = (newValue: string) => {
+    if (newValue.length > maxHeadingLength) {
+      setIsMaxHeadingLengthExceeded(true);
+      setTimeout(() => {
+        setIsMaxHeadingLengthExceeded(false);
+      }, 2000);
+      return;
+    }
+    setHeading(newValue);
+  };
 
   return (
-    <h1
-      className={cn(className, props.className)}
-      ref={ref}
-      contentEditable
-      suppressContentEditableWarning={true}
-      onBlur={(e) => {
-        const text = e.currentTarget.innerHTML;
-        if (text !== props.text) {
-          props.onUpdated(text);
-        }
-      }}
-      onKeyDown={(e) => {
-        if (e.key === 'Enter' && ref.current) {
-          ref.current.blur();
-        }
-      }}
-    >
-      {props.text}
-    </h1>
+    <div>
+      <textarea
```

Comment:
An `h1` is a better element for this purpose in terms of semantics. Can you help me understand why that won't work here? What are the problems with using an h1 and are there ways to overcome them?

Reply:
I'm not yet convinced we need to change to using a textarea instead of an h1. If it is the case that we are unable to prevent editing a contenteditable element when the length exceeds N, then we should use an h1 until they focus the element for editing at which point we can change to a textarea. Then we should change it back on blur.

I'll take a quick look at this tomorrow.